test/fixtures/dhcp/dhcp.leases

```diff
@@ -13,19 +13,33 @@ host ten.example.com {
 }
 
 # Reservation that is deleted
-host deleteme.example.com {
+host deleted.example.com {
   dynamic;
   hardware ethernet 10:10:10:10:20:20;
   fixed-address 192.168.122.20;
   supersede host-name = "deleteme.example.com";
 }
-host deleteme.example.com {
+host deleted.example.com {
+  dynamic;
+  deleted;
+}
+# Reservation that is deleted and then undeleted with same name
+host undeleted.example.com {
+  dynamic;
+  hardware ethernet 10:10:10:10:30:30;
+  fixed-address 192.168.122.30;
+  supersede host-name = "undeleted.example.com";
+}
+host undeleted.example.com {
   dynamic;
-  hardware ethernet 10:10:10:10:20:20;
-  fixed-address 192.168.122.20;
-  supersede host-name = "deleteme.example.com";
   deleted;
 }
+host undeleted.example.com {
+  dynamic;
+  hardware ethernet 10:10:10:10:30:30;
+  fixed-address 192.168.122.30;
```

Comment:
Could you make this a different IP, and ensure in the tests that only the new IP is in use, and that the prior IP is available?